subprojects/core-api/src/main/java/org/gradle/api/NamedDomainObjectCollection.java

```diff
@@ -216,7 +230,8 @@ public interface NamedDomainObjectCollection<T> extends DomainObjectCollection<T
      * Returns a collection containing the objects with names matching the provided filter.
      * The returned collection is live, so that when matching objects are added to this collection,
      * they are also visible in the filtered collection.
-     * This method will NOT cause any pending objects in this container to be realized.
+     * <p>
+     * This method is an intermediate operation. It is neither lazy nor eager by itself.
```

Comment:
"neither lazy nor eager" is confusing, what does that mean?

Reply:
That it all depends on what you do with the returned collection.

How would you express that? 
I was hoping that using "intermediate" would help users make a parallel with Java streams.